internal/services/securitycenter/advanced_threat_protection_resource.go

```diff
@@ -88,6 +88,34 @@ func resourceAdvancedThreatProtectionCreateUpdate(d *pluginsdk.ResourceData, met
 		return fmt.Errorf("updating Advanced Threat protection for %q: %+v", id.TargetResourceID, err)
 	}
 
+	deadline, _ := ctx.Deadline()
+
+	// the API appears to be eventually consistent, tracked on https://github.com/Azure/azure-rest-api-specs/issues/25232
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending: []string{"diff"},
+		Target:  []string{"consistent"},
+		Refresh: func() (result interface{}, state string, err error) {
+			resp, err := client.Get(ctx, id.TargetResourceID)
+			if err != nil {
+				return resp, "error", err
+			}
+			if atpp := resp.AdvancedThreatProtectionProperties; atpp != nil {
+				if atpp.IsEnabled != nil && *atpp.IsEnabled == d.Get("enabled").(bool) {
```

Comment:
`nil` and `false` here could be the same thing? Are we certain that in the not enabled case we always get a value back here?

Reply:
I'm not sure, taking `nil` as `false` does make sense.